src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.kt

```diff
@@ -230,6 +244,36 @@ class AttendanceService(
     telemetryClient.trackEvent("OffenderDelete", mapOf("offenderNo" to offenderNo, "count" to  totalAttendances.toString()), null)
   }
 
+  fun getAttendanceChanges(): Set<AttendanceChangeDto> {
+    val changes = attendanceChangesRepository
+        .findAll()
+        .toSet()
+
+    val attendanceIds = changes
+        .map { it.attendanceId }
+        .toSet()
+
+    val attendancesMap = attendanceRepository
+        .findAllById(attendanceIds)
+        .associate { Pair(it.id, it) }
+
+    return changes.map {
+      val attendance = attendancesMap.get(it.attendanceId)!!
+
+      AttendanceChangeDto(
+          id = it.id!!,
+          attendanceId = it.attendanceId,
+          bookingId = attendance.bookingId,
+          eventId = attendance.eventId,
+          eventLocationId = attendance.eventLocationId,
+          changedFrom = it.changedFrom,
+          changedTo = it.changedTo,
+          changedOn = it?.createDateTime,
+          changedBy = it?.createUserId
```

Comment:
this right here is pretty genious

Reply:
lol Thanks. However, I might end of changing some of it as I really should let JPA deal with the change->attendance relationship.